models/MigrationService.cfc

```diff
@@ -2,22 +2,83 @@ component singleton accessors="true" {
 
 	property name="wirebox" inject="wirebox";
 	property name="migrationsDirectory" default="/resources/database/migrations";
-	property name="datasource";
-	property name="defaultGrammar"  default="AutoDiscover@qb";
-	property name="schema"          default="";
-	property name="migrationsTable" default="cfmigrations";
+	property name="seedsDirectory"      default="/resources/database/seeds";
+	property name="manager"             default="cfmigrations.models.QBMigrationManager";
+
+
+	MigrationService function init() {
+		structAppend( variables, arguments, true );
+		return this;
+	}
+
+	function onDIComplete() {
+		variables.manager = variables.wirebox.getInstance( variables.manager );
+
+		var omit = [
+			"wirebox",
+			"migrationsDirectory",
+			"seedsDirectory",
+			"manager"
+		];
+
+		var managerProperties = getMetadata( variables.manager ).properties.map( function( prop ) {
+			return prop.name;
+		} );
+
+		variables
```

Comment:
Effectively, the managers need different variables in the configuration.  This is the way to seamlessly pass down custom variables ( other than the service's own variables ).  So, for example, the QB Migration manager might need the grammar, but the MongoDB would blow up with that setter being called, so this allows, for example the `migrationsCollection` to be passed through to that manager.

Reply:
I think you are overcomplicating the approach to pass in properties to manager. Why call concrete setters when you can just seed the protocols with the properties collection. Let each manager, do what it needs with the `properties` instead of individual settings from the collection.  This is overkill in my opinion. When all you need to is to seed in the properties.

Reply:
@lmajano : Could you give me an example, using your approach?  You mean nested structs and just inject the properties?

Reply:
I'm going to go on that assumption.  Doing so, though means a config struct that is different than the current, which I'm fine with - but we're going to have to code for backward compat.  Here's my suggestion for a new config struct:

```
moduleSettings = {
    "cfmigrations" : {
         "manager" : "QBMigrationManager@cfmigrations",
         "properties" : {
                 "migrationsDirectory" : "/resources/database/db1/migrations",
                "defaultGrammar" : "MySQLGrammar@qb",
                "datasource" : "db1",
                "useTransactions" : "false"
         }
    }
}
```

If we do this, I can just pass them in directly as init arguments and it's straight forward, with the understanding that we'll  have to write a workaround to handle the current flatted ModuleConfig settings struct

Reply:
I read your changes to the README and I think we can go for a breaking change and just use a map of `managers`.  We can have the default be a single `QBManager@cfmigrations` pointing at `/resources/database/migrations` and using the default datasource.

Reply:
I do like the `properties` key, though.  Like interceptors or cbmailservices providers, I think having a `manager` key and `properties` key is all we allow.  `properties` is where anything goes and is `manager` dependent.

Reply:
OK. I've converted the settings structure to the `manager`/`properties` separation.  The README has been updated to reflect the new conventions for single or multiple managers.